src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorPromise.cs

```diff
@@ -83,9 +83,9 @@ public ExportDescriptor GetDescriptor()
             _creating = true;
             try
             {
-                var reply = _descriptor.Value;
-                Assumes.IsTrue(reply != null, "Export descriptor fulfillment function returned null.");
-                return reply;
+                ExportDescriptor relay = _descriptor.Value;
+                Requires.NotNull(relay, "getDescriptor");
```

Comment:
I don't think we should keep that exception. The exception can't be caught anyways as it's internal so the only thing that can catch this is a global catch block. So the new ANE will also be caught anyway.